### DISCUSSION TITLE
Budget Fixed amount not working

### DISCUSSION THREAD

**Discussion by ghost** [firefly-iii/firefly-iii#6238] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Fixed amount in Budget does not automatically set the budget amount to each month.

[https://docs.firefly-iii.org/firefly-iii/concepts/budgets/](url)
`Firefly III will create a new budget limit every selected period. The exact behavior depends entirely on your settings:
If you set it to "monthly", € 200,- Firefly III will give you an automatic budget limit of € 200,-, valid for one month, every month. This will happen automatically.`

Am I doing something wrong?
I set date from Jan 1st to Dec 31st, set to Fixed amount and amount, then save.

The line for the budget appears but no budgeted amount is automatically setup.
[https://nsm09.casimages.com/img/2022/07/17//22071705055125632817951997.png](url)

### Expected behaviour

Amount should be set automatically when using Fixed Amount, see desc.

### Debug information

Debug information generated at 2022-07-17 16:39:15 Europe/Madrid for Firefly III version **5.7.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.9 |
| Firefly III API | 1.5.6 |
| Build | 196, 09-06-2022 18:30:26 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.0.17` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Madrid |
| Browser TZ | America/Toronto |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-01-01 00:00:00 |
| Session end | 2023-12-31 00:00:00 |
| Session first | 2022-01-01 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36 |

### Steps to reproduce

Time Frame is "January 1st, 2022 - December 31st, 2023"
[https://nsm09.casimages.com/img/2022/07/17//22071705161825632817952004.png](url)
1. "New Budget"
2. write the name (ex. Rent)
3. set Auto-Budget to "Set a fixed amount every period"
4. set the amount (ex. 650$)
5. Auto-budget period to "Monthly" --> [https://nsm09.casimages.com/img/2022/07/17//22071705183325632817952006.png](url)
6. "Sotre new budget"
Result --> [https://nsm09.casimages.com/img/2022/07/17//22071705055125632817951997.png](url)

**Comment by JC5** [comment]
What's the cron job log look like?

**Comment by ghost** [comment]
> What's the cron job log look like?

Where to I find that info? I'm unraid server.

**Comment by JC5** [comment]
A cron job is something you have to set up yourself. I know very little about the Unraid platform, but there are instructions in the documentation.

https://docs.firefly-iii.org/firefly-iii/advanced-installation/cron/

**Reply by ghost** [reply]
Hopefully, that features change to something a lot easier to use in a near future! 😉
